monai/data/box_utils.py

```diff
@@ -0,0 +1,173 @@
+# Copyright (c) MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from copy import deepcopy
+from typing import Sequence, Union
+
+import numpy as np
+import torch
+
+from monai.config.type_definitions import NdarrayOrTensor
+from monai.data import box_mode
+from monai.data.box_mode import (
+    BoxMode,
+    CenterSizeMode,
+    CornerCornerModeTypeA,
+    CornerCornerModeTypeB,
+    CornerCornerModeTypeC,
+    CornerSizeMode,
+)
+from monai.utils import look_up_option
+from monai.utils.enums import BoundingBoxMode
+from monai.utils.type_conversion import convert_data_type, convert_to_dst_type
+
+# TO_REMOVE = 0 if the bottom-right corner pixel/voxel is not included in the box,
+#      i.e., when xmin=1, xmax=2, we have w = 1
+# TO_REMOVE = 1  if the bottom-right corner pixel/voxel is included in the box,
+#       i.e., when xmin=1, xmax=2, we have w = 2
+# Currently, only `TO_REMOVE = 0.` is supported
+TO_REMOVE = box_mode.TO_REMOVE
+
+# We support the conversion between several box modes, i.e., representation of a bounding box
+# BOXMODE_MAPPING maps string box mode to the corresponding BoxMode class
+BOXMODE_MAPPING = {
+    BoundingBoxMode.XYXY: CornerCornerModeTypeA(),  # [xmin, ymin, xmax, ymax]
+    BoundingBoxMode.XYZXYZ: CornerCornerModeTypeA(),  # [xmin, ymin, zmin, xmax, ymax, zmax]
+    BoundingBoxMode.XXYY: CornerCornerModeTypeB(),  # [xmin, xmax, ymin, ymax]
+    BoundingBoxMode.XXYYZZ: CornerCornerModeTypeB(),  # [xmin, xmax, ymin, ymax, zmin, zmax]
+    BoundingBoxMode.XYXYZZ: CornerCornerModeTypeC(),  # [xmin, ymin, xmax, ymax, zmin, zmax]
+    BoundingBoxMode.XYWH: CornerSizeMode(),  # [xmin, ymin, xsize, ysize]
+    BoundingBoxMode.XYZWHD: CornerSizeMode(),  # [xmin, ymin, zmin, xsize, ysize, zsize]
+    BoundingBoxMode.CCWH: CenterSizeMode(),  # [xcenter, ycenter, xsize, ysize]
+    BoundingBoxMode.CCCWHD: CenterSizeMode(),  # [xcenter, ycenter, zcenter, xsize, ysize, zsize]
+}
+# The standard box mode we use in all the box util functions
+StandardMode = CornerCornerModeTypeA
+
+
+def get_boxmode(mode: Union[str, BoxMode, None] = None) -> BoxMode:
```

Comment:
This `mapping` logic is duplicated to the `name` variable in every `BoxMode` class.
I think maybe we can change this overall logic base on my above comments:
```py
SUPPORTED_SPATIAL_DIMS = [2, 3]

SUPPORTED_MODES: [
    CornerCornerModeTypeA,
    CornerCornerModeTypeB,
    CornerCornerModeTypeC,
    CornerSizeMode,
    CenterSizeMode,
]

def get_boxmode(mode: Union[str, BoxMode, Type[BoxMode], None] = None, *args, **kwargs) -> BoxMode:
    if isinstance(mode, BoxMode):
        return mode
    if issublcass(mode, BoxMode):
        return mode(*args, **kwargs)
    elif isinstance(mode, str):
        for m in SUPPORTED_MODES:
            for n in SUPPORTED_SPATIAL_DIMS:
                if m.get_name(n) == mode
                    return m(*args, **kwargs)
    elif mode is None:
        return StandardMode(*args, **kwargs)
    raise ValueError(f"unsupported box mode: {mode}.")
```

What do you think?

Thanks.